src/patternfly/layouts/Page/styles.scss

```diff
@@ -1,110 +1,307 @@
+
+      @media screen and (min-width: $pf-global--breakpoint--lg) {
+        flex: 1;
+        order: initial;
+        width: auto;
+        padding: 0;
+        margin-right: var(--pf-l-page__header-nav--MarginRight--md);
+        margin-left: var(--pf-l-page__header-nav--MarginLeft--md);
+        background-color: transparent;
+      }
+    }
+
+    &-selector {
+      flex-shrink: 1;
+      align-items: center;
+      padding-top: var(--pf-l-page__header-selector--PaddingTop);
+      padding-bottom: var(--pf-l-page__header-selector--PaddingBottom);
+
+      @media screen and (min-width: $pf-global--breakpoint--md) {
+        padding-left: var(--pf-l-page__header-selector--PaddingLeft--md);
+      }
     }
 
     &-tools {
+      align-items: center;
       margin-left: auto; // to push it to the right
+
+      .pf-l-toolbar {
+        margin-top: var(--pf-l-page__header-tools--toolbar--MarginTop);
+        margin-bottom: var(--pf-l-page__header-tools--toolbar--MarginBottom);
+      }
     }
 
-    &-nav {
-      // this is not mobie first since we are changing the order and it'd be too much overwrite to do it with a min-width
-      @media (max-width: $pf-global--breakpoint--md) {
-        flex: 1 0 100%;
-        order: 1;
-        background-color: var(--pf-l-page__header-nav--sm--BackgroundColor);
+    &-avatar {
+      width: var(--pf-l-page__header-avatar--Height);
+      height: var(--pf-l-page__header-avatar--Width);
+
+      .pf-l-toolbar + & {
+        margin-left: var(--pf-l-page__header-avatar--MarginLeft);
       }
     }
+  }
+
+  // Sidebar
+  &__sidebar {
+    grid-area: 2 / 1;
+    z-index: var(--pf-l-page__sidebar--ZIndex);
+    width: var(--pf-l-page__sidebar--Width--sm);
+    padding-top: var(--pf-l-page__sidebar--PaddingTop);
+    padding-bottom: var(--pf-l-page__sidebar--PaddingBottom);
+    overflow: hidden;
+    background-color: var(--pf-l-page__sidebar--BackgroundColor);
+    transition: var(--pf-l-page__sidebar--Transition);
+    transform: var(--pf-l-page__sidebar--Transform--out);
+
+    @media screen and (min-width: $pf-global--breakpoint--md) {
+      flex: 1;
+      order: initial;
+      width: var(--pf-l-page__sidebar--Width--lg);
+      // Max-width value set for browser to calculate max-width while transitioning
+      max-width: var(--pf-l-page__sidebar--MaxWidth--lg);
+      box-shadow: var(--pf-l-page__sidebar--BoxShadow);
+      transform: var(--pf-l-page__sidebar--Transform--in);
+    }
 
-    @media (min-width: $pf-global--breakpoint--md) {
-      min-height: pf-size-prem(110px);
+    // Mobile
+    // Expanded nav
+    &.pf-m-expanded {
+      box-shadow: var(--pf-l-page__sidebar--BoxShadow);
+      transform: translate3d(0, 0, 0);
+    }
+
+    // Desktop
+    // Collapse nav
+    &.pf-m-collapsed {
 
-      &-brand {
-        background-color: var(--pf-l-page__header-brand--md--BackgroundColor);
+      @media screen and (min-width: $pf-global--breakpoint--md) {
+        max-width: 0;
```

Comment:
Transitioning `translate` (like `transform: translateX(-100%)`) is more performant/less choppy. You could use that instead of `max-width` here pretty easily. Not sure if you could also do something similar on `.pf-l-page__main`? Might be worth playing around with.